tests/test_unittest/test_merge_blocks_pass.py

```diff
@@ -0,0 +1,322 @@
+    ij_offset: Tuple[int, int, int],
+    ijk_domain: Domain,
+) -> None:
+    transform_data = make_transform_data(
+        name="ij_offset_readonly",
+        domain=ijk_domain,
+        fields=["out", "in", "inout"],
+        body=[("out", "in", ij_offset), ("inout", "in", (0, 0, 0))],
+        iteration_order=iteration_order,
+    )
+    transform_data = merge_blocks_pass(transform_data)
+    #  allowed to be merged: no write to "in"
+    assert len(transform_data.blocks) == 1
+
+
+def test_merge_write_after_read_k(
+    merge_blocks_pass: PassType, non_parallel_iteration_order: IterationOrder, ijk_domain: Domain
+) -> None:
+    transform_data = make_transform_data(
+        name="k_offset_nonparallel",
+        domain=ijk_domain,
+        fields=["out", "inout", "in"],
+        body=[("out", "inout", (0, 0, -1)), ("inout", "in", (0, 0, 0))],
+        iteration_order=non_parallel_iteration_order,
+    )
+    transform_data = merge_blocks_pass(transform_data)
+    # allowed to be merged: order is sequential -> no race condition on k-offsets
+    assert len(transform_data.blocks) == 1
+
+
+def test_merge_write_after_read_k_parallel(
+    merge_blocks_pass: PassType, ijk_domain: Domain
+) -> None:
+    transform_data = make_transform_data(
+        name="k_offset_parallel",
+        domain=ijk_domain,
+        fields=["out", "inout", "in"],
+        body=[("out", "inout", (0, 0, -1)), ("inout", "in", (0, 0, 0))],
+        iteration_order=IterationOrder.PARALLEL,
+    )
+    transform_data = merge_blocks_pass(transform_data)
+    # not allowed to be merged: order is PARALLEL -> race condition even on k-offsets
+    assert len(transform_data.blocks) == 2
+
+
+def test_merge_write_after_read_k_extended_sequential(
+    merge_blocks_pass: PassType, non_parallel_iteration_order: IterationOrder, ijk_domain: Domain
+) -> None:
+    transform_data = make_transform_data(
+        name="k_extended",
+        domain=ijk_domain,
+        fields=["out", "in", "inout"],
+        body=[("tmp", "inout", (0, 0, 0)), ("out", "tmp", (0, 0, 1)), ("inout", "in", (0, 0, 0))],
+        iteration_order=non_parallel_iteration_order,
+    )
+    transform_data = merge_blocks_pass(transform_data)
+    # should be merged, since k-offset in sequential order does not extend compute domain
+    assert len(transform_data.blocks) == 1
+
+
+def test_merge_read_after_write_k_parallel_seq(
+    merge_blocks_pass: PassType, ijk_domain: Domain
+) -> None:
+    transform_data = make_transform_data(
+        name="read_after_write_forbidden",
+        domain=ijk_domain,
+        fields=["out", "in"],
+        body=[("tmp", "in", (0, 0, 0)), ("out", "tmp", (0, 0, -1))],
+        iteration_order=IterationOrder.PARALLEL,
+    )
+    transform_data = merge_blocks_pass(transform_data)
+    # not allowed to be merged, because PARALLEL and k-axis is sequential
+    assert len(transform_data.blocks) == 2
+
+
+@pytest.mark.skip(reason="ComputeExtentsPass fails if no sequential axis")
```

Comment:
It is a (very low priority) bug. I will create an issue for it.
It seems that in the wild there are no domains without sequential axis. As such it may considered to be not a bug, however then we should stop supporting that case elsewhere and not allow it to exist in the first place.

Reply:
It is technically a bug. Lack of support for a case that is supported elsewhere but does not seem to have a usecase currently.

I will create an issue for this.